metrics_sdk/lib/opentelemetry/sdk/metrics/meter_provider.rb

```diff
@@ -0,0 +1,121 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module SDK
+    # The Metrics module contains the OpenTelemetry metrics reference
+    # implementation.
+    module Metrics
+      # {MeterProvider} is the SDK implementation of {OpenTelemetry::Metrics::MeterProvider}.
+      class MeterProvider < OpenTelemetry::Metrics::MeterProvider
+        Key = Struct.new(:name, :version)
+        private_constant(:Key)
+
+        def initialize(resource: OpenTelemetry::SDK::Resources::Resource.create)
+          @mutex = Mutex.new
+          @registry = {}
+          @stopped = false
+
+          @metric_readers = []
+          @views = []
+        end
+
+        # Returns a {Meter} instance.
+        #
+        # @param [String] name Instrumentation package name
+        # @param [optional String] version Instrumentation package version
+        #
+        # @return [Meter]
+        def meter(name, version = nil)
```

Comment:
Are we holding off on the `schema_url` for now?

Reply:
Yeah, I'd like to.  Still needs to be added to the tracing portion of things.  May as well do it at the same time as one consistent PR.